generative_data_prep/processors/article_tokenizer.py

```diff
@@ -131,10 +137,37 @@ def __call__(self, article: Optional[str]) -> List[TokenizedSequence]:
             err_msg += f" must be {FileExtension.JSONL} or {FileExtension.TXT}"
             raise ValueError(err_msg)
 
+        # Check if any article in tokenized_articles is all prompts
+        # Q: should we do this? or just check after packing?
+        if not self.keep_prompt_only_sequences:
+            tokenized_articles = self._remove_prompt_only_sequences(tokenized_articles)
+
         tokenized_sequences = self.packer(tokenized_articles)
 
+        # Check if any sequence in tokenized_sequences is all prompts
+        if not self.keep_prompt_only_sequences:
+            return self._remove_prompt_only_sequences(tokenized_sequences)
         return tokenized_sequences
 
+    def _remove_prompt_only_sequences(self, tokenized_lines: List[TokenizedLine]) -> List[TokenizedLine]:
+        """Takes a list of TokenizedLines, removes those that don't contain any COMPLETION TokenTypeIds.
+
+        Args:
+            tokenized_lines: List of TokenizedLines
+        Returns:
+            Original list with prompt-only sequences filtered out
+        
+        """
+        filtered_lines = []
+        for line in tokenized_lines:
+            if not any([token_type == TokenTypeIds.COMPLETION for token_type in line.token_type_ids()]):
+                if not self.logged_prompt_only_warn_msg:
+                    print('WARNING: --keep_prompt_only_sequences is not set and sequences with no COMPLETION TokenTypeIds will be thrown away. Will only print this warning once.')
```

Comment:
More clear error message.

After packing data into sequences, some sequences contain ONLY prompt tokens. These sequences and tokens are being discarded